### PR TITLE
Fix parsing errors

### PR DESCRIPTION
**Pull request by bryan-hoang**
I didn't thoroughly test the plugin on a large code base, which led to spurious errors from the plugin failing to parse edge cases (e.g., failing to parse `const path = require('node:path');`). To address the issues, I decided to improve the testing DX by switching from `mocha` to `vitest` before moving on to tackling the issues.

In addition to adding multiple test cases and CI checks, I opted to use https://github.com/eslint/eslint/blob/main/lib/rules/brace-style.js as a reference to modify it to fit the brace style of the company's style guide.

### CODE REVIEW

**Comment by bryan-hoang** [comment]
@j0ashm Feel free to skim it to see the changes if ya have the time. I should've done a better job testing the plugin on our monorepo 😅.

**Comment by j0ashm** [comment]
@bryan-hoang Out of curiosity, what kind of edge cases were missed? Reading the issue, I'm not quite sure where/why `node:path` comes into play as a parsing error?

**Comment by j0ashm** [comment]
I do remember running the plug-in on the DCP monorepo and it didn't throw up any errors... maybe I had it misconfigured?

**Comment by bryan-hoang** [comment]
> @bryan-hoang Out of curiosity, what kind of edge cases were missed? Reading the issue, I'm not quite sure where/why `node:path` comes into play as a parsing error?

@j0ashm If you're on the `main` branch and add `const path = require('node:path');` as a test cast, the test will fail. e.g., apply [test-case.patch](https://github.com/Distributive-Network/eslint-plugin-distributive/files/12450285/test-case.patch) and run the tests on `main`.

> I do remember running the plug-in on the DCP monorepo and it didn't throw up any errors... maybe I had it misconfigured?

After linking the plugin & config in, I essentially ran `npx -- eslint --ignore-path=.gitignore --ext=js,simple,tap,simple.failing,tap.failing .`, and ran into the error on `attic/bank/bin/bank.js:3`, which is the failing test case.

File: .eslintrc.js [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
     node: true,
-    es6: true,
-    browser: true
   },
-  overrides: [
-    {
-      files: ["tests/**/*.js"],
-      env: { mocha: true },
+  overrides: [ {
+    files: ['**/*.mjs'],
```
Comment (edeluzio) [inline .eslintrc.js:25]: Just wanna make sure this was intentional, switching overrides from .js to .mjs. Why do we do this?

File: docs/rules/brace-style.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+};
+```
+
+## When Not To Use It
+
+If you don't plan on following this brace style.
+
+## Further Reading
+
+- <https://github.com/Distributive-Network/eslint-config#braces>
```
Comment (edeluzio) [inline docs/rules/brace-style.md]: (Nitpick) Page has two of these links, one at the top and one here at the bottom. I'd vote to choose one of them, unless there's a good reason for having both

**Review by edeluzio — COMMENTED** [review]
A couple nitpicks, but overall LGTM. Running without problems locally

File: .eslintrc.js (hunk 1 above)
Comment (bryan-hoang) [reply .eslintrc.js:25]: Yep! Since the [`"types"`](https://nodejs.org/api/packages.html#type) field of the `package.json` file is omitted, all `.js` files by default are CommonJS modules when loaded by Node. But Vitest's config file format has to be an ECMAScript Module, so the `.mjs` extension tells Node.js to load it as an ES module.

ESLint needs to be configured to understand that behaviour, unfortunately. It'd be nice if it understood that behaviour by default.

File: docs/rules/brace-style.md (hunk 2 above)
Comment (bryan-hoang) [reply docs/rules/brace-style.md]: @edeluzio I've made the second link refer to the README in general, rather than the same section linked twice. I'd like to keep the first link, as it's meant to explain where the rule comes from, whereas further reading would be for folks interested in learning more, I suppose.

For whatever reason, GitHub isn't updating this PR with my recent pushes :/

**Comment by edeluzio** [comment]
Good change with the link, it's minor but better IMO.
LGTM!

**Review by edeluzio — COMMENTED** [review]
LGTM!

**Review by edeluzio — APPROVED** [review]

**Review by eroosenmaallen — APPROVED** [review]
Looks right
